src/main/cpp/utils/StateManager.cpp

```diff
@@ -0,0 +1,52 @@
+#include "utils/StateManager.h"
+
+template <typename StateType>
+StateManager<StateType>::StateManager(const std::function<bool()> Interrupt,
+                                      StateType DefaultState,
+                                      frc2::CommandPtr DefaultEntrypoint)
```

Comment:
All of these `frc2::CommandPtr` should be r-values (`frc2::CommandPtr&&`)

Reply:
if I make it take CommandPtr&& can I do away with the std::move?

Reply:
You shouldn't have to if the map's move overload is being used on assignment, but double-check that

Reply:
Attempted this, it didn't seem to want to build. I assume it's my inexperience with pointers

Reply:
Need an `std::move`